File: 2580-circular-sentence/circular-sentence.java
class Solution {
    public boolean isCircularSentence(String sentence) {
        char start = sentence.charAt(0);
        // char last = "";
        String[] s = sentence.split(" ");
        
        for(int i = 0; i < s.length; i++){
            char last = s[i].charAt(s[i].length() - 1);
            if(i != s.length - 1){
                char first = s[i + 1].charAt(0);
                
                if(last != first){
                    return false;
                }
            }else{
                if(last != start){
                    return false;
                }
            }
        }
        System.out.println(Arrays.toString(s));
        
        return true;
    }
}

File: 2580-circular-sentence/circular-sentence.py
class Solution:
    def isCircularSentence(self, sentence: str) -> bool:
        l=sentence.split()
        c=0
        z=l[-1]
        y=l[0]
        print(z)
        print(y)
        
        if z[-1]==y[0]:
            c=1
            if (z.islower() and y.isupper()) or (z.isupper() and y.isupper()):
                c=0
            
        
            
            for i in range(len(l)-1):
                a=l[i]
                b=l[i+1]
                if a[-1]==b[0]:
                    c=1
                else:
                    c=0
                    break;
        else:
            c=0
       
        if c==1:
            return True
        else:
            return False
        